feat!: remove sha256 opcode

guipublic opened the pull request:
This PR resolves Noir issue 4330: https://github.com/noir-lang/noir/issues/4330
by removing the sha256 opcode and replacing the sha256 function in the stdlib by the implementation using the sha256 compression opcode (also in the stdlib).

guipublic committed:
Remove sha256 opcode

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix test case

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
remove sha256 from acvm_js

guipublic committed:
fix the build for acvm-js

guipublic committed:
fix for ecdsa account contract

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
remove sha256 test case from acvm_js

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
remove sha256 from noir_js import

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix test case

kevaundray committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
merge from master

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix the build

guipublic committed:
fix the build(2)

guipublic committed:
Fix issue with conditional_1 test case

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix contract for sha256

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix merge from master

guipublic committed:
update test snapshots

guipublic committed:
fix dsl unit test

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
add sha_256_compression in acvm_js blackbox

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
update test snapshot

guipublic committed:
export sha256_compression in acvm_js

guipublic committed:
fix acvm_js build

guipublic committed:
cargo fmt

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
update snapshot

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
update artifact hash

TomAFrench committed:
chore: move playwright install before building yarn packages

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330
* master:
  fix: Broadcasting unconstrained function with empty sibling (#5429)
  feat: Sync from noir (#5416)
  refactor: Meld flavor and and circuit builder modules (#5406)
  feat: less earthly runners + e2e GA runners, bb bench (#5356)
  fix: Track class registered count in tx stats (#5417)

TomAFrench commented:
artifact hash is likely wrong and snapshots will need to be updated I think.

TomAFrench committed:
chore: fix version of playwright

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: update playwright in yarn.lock

TomAFrench committed:
chore: fix formatting

TomAFrench committed:
chore: more formatting

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: fix acvm_js test

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: update serializer

guipublic committed:
update acvm_js test serialization

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: update hash

TomAFrench committed:
chore: update deployer address

TomAFrench committed:
chore: fix deployer address

TomAFrench commented on noir/noir-repo/noir_stdlib/src/hash.nr:
```diff
@@ -6,11 +6,12 @@ mod pedersen;
+    crate::sha256::digest(input)
+}
 
 #[foreign(sha256)]
```
This is still calling the old blackbox function.

guipublic replied:
sha256() is not calling the old one, or do you mean sha256_slice()?

TomAFrench replied:
Yep, I linked to the line with `#[foreign(sha256)]` on it. It's just showing a few lines above as well.

guipublic replied:
I believe we do not support hash of slices, do we?
I'd rather remove it then.

TomAFrench replied:
It seems like we do support it based on the other functions below.

guipublic replied:
We should discuss about the removal of hash of slices.

TomAFrench commented on noir/noir-repo/compiler/noirc_evaluator/src/ssa/ir/instruction.rs:
```diff
@@ -262,6 +262,8 @@ impl Instruction {
             | Load { .. }
             | Store { .. }
             | IncrementRc { .. }
+            | ArrayGet { .. }
+            | ArraySet { .. }
```
Why was this change made?

guipublic committed:
Code review

guipublic commented on noir/noir-repo/noir_stdlib/src/field.nr:
```diff
@@ -47,6 +47,24 @@ impl Field {
+            result[i] = bytes[i];
+        }
+        result
+    }
```
Done

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330
* master: (92 commits)
  feat(acir)!: Add predicate to call opcode (#5616)
  feat: only export values from accumulated data (#5604)
  feat: Improve the proving orchestration lifecycle (#5535)
  chore(docs): Random updates (#5281)
  feat: `add` and `sub` methods of `EasyPrivateUint` throw when called in public (#5581)
  Update quickstart.md (#5605)
  chore(docs): Fix indexed-merkle-tree docs images (#4674)
  feat: Sync from noir (#5572)
  feat: /foundry is canoncial build of foundry. e2e tests can start own anvil. (#5522)
  fix: remove EFS lifecycle rule (#5587)
  fix: invalid fork terraform (#5585)
  fix(ci): Install fixed foundry version in CI (#5582)
  docs(spec): hashing and keys (#5478)
  fix: mainnet fork redeploys (#5573)
  git subrepo push --branch=master noir-projects/aztec-nr
  git_subrepo.sh: Fix parent in .gitrepo file. [skip ci]
  chore: replace relative paths to noir-protocol-circuits
  git subrepo push --branch=master barretenberg
  refactor: nuking accounts from e2e setup (#5574)
  feat(avm): Track gas from memory accesses explicitly (#5563)
  ...

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix merge issue

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: remove sha256 from avm-transpiler

TomAFrench committed:
Apply suggestions from code review

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

Commit:
chore(avm-simulator): sha256 -> sha256compression

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge remote-tracking branch 'origin/db/avm-simulator-sha256compression' into gd/noir_issue_4330

guipublic committed:
update serialization in unit tests

guipublic committed:
test

guipublic committed:
revert test

guipublic committed:
update acir serialization format

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
update serialisation

guipublic committed:
another serialisation update

guipublic committed:
fix merge issue

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
update acir.cpp

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
re-generate serialisation format

guipublic committed:
fix merge issue

guipublic committed:
fix merge issue

guipublic committed:
fix merge issues

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: fm

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

dbanks12 committed:
chore(avm-simulator): sha256 -> sha256compression (#6067)

Please read [contributing guidelines](CONTRIBUTING.md) and remove this
line.

---------

Co-authored-by: Tom French <15848336+TomAFrench@users.noreply.github.com>
Co-authored-by: guipublic <47281315+guipublic@users.noreply.github.com>

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: remove file imports with portals

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330
* master: (37 commits)
  refactor: nuking `KeyStore` and `KeyPair` interfaces (#6553)
  feat: Sync from noir (#6555)
  refactor: purge secret and open keywords (#6501)
  feat: run benchmarks for ACIR proving (#6155)
  feat: generic key validation request (#6474)
  chore: fix migration notes (#6551)
  git subrepo push --branch=master noir-projects/aztec-nr
  git_subrepo.sh: Fix parent in .gitrepo file. [skip ci]
  chore: replace relative paths to noir-protocol-circuits
  git subrepo push --branch=master barretenberg
  feat: Adding autogenerated variants for the reset circuit (#6508)
  chore: Copy subset of constants to cpp (#6544)
  feat(docs): Key rotation / owner -> nullifier key docs (#6538)
  chore(docs): restructure improvs (#6502)
  chore: anvil kill wrapper now supports mac (#6520)
  feat: sum transaction fees and pay on l1 (#6522)
  refactor: private call validation (#6510)
  chore: add c++ tests for generator derivation (#6528)
  feat: View functions with static context enforcing (#6338)
  feat: Update the encrypted note log format (#6411)
  ...

guipublic committed:
use no_predicate attribute for sha256

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: remove references to `sha256_constraint`

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: fix CI issue

TomAFrench reviewed (changes requested):
blocking due to release freeze. Merging this would prevent syncs

TomAFrench committed:
chore: ignore slow debugger test

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix merge issue

TomAFrench reviewed (approved):
The release freeze has ended so this can go in now.

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
revert yarn.lock

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
chore: update yarn.lock

guipublic committed:
fix merge issues #2

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix merge issue #3

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
format

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330
* master: (176 commits)
  refactor: counters (#7342)
  chore: notify Noir team on changes to the stdlib (#7387)
  chore(bb): do not instantiate Relation (#7389)
  chore(avm): avoid including flavor where possible (#7361)
  chore(avm): better log_derivative_inverse_round (#7360)
  chore(avm): smaller prover (#7359)
  chore(avm): smaller transcript (#7357)
  feat: Sync from noir (#7352)
  feat: VK tree (#6914)
  chore(docs): Update token tutorial (#7241)
  chore: sync external contributions (#6984)
  fix: privacy leak in private refund (#7358)
  fix: included argshash computation in public call_interfaces and cleanup (#7354)
  docs: making private refunds smooth-brain friendly (#7343)
  refactor: optimize public call stack item hashing (#7330)
  fix: Allow importing notes from other contracts and inject them in the macros (#7349)
  refactor: Optimize private call stack item hash for gate count (#7285)
  git subrepo push --branch=master noir-projects/aztec-nr
  git_subrepo.sh: Fix parent in .gitrepo file. [skip ci]
  chore: replace relative paths to noir-protocol-circuits
  ...

TomAFrench committed:
Merge branch 'master' into gd/noir_issue_4330
* master:
  feat!: constant inputs for blackbox (#7222)

TomAFrench committed:
chore: clippy

TomAFrench committed:
Update noir/noir-repo/noir_stdlib/src/hash/mod.nr

TomAFrench committed:
.

TomAFrench committed:
.

TomAFrench committed:
.

TomAFrench committed:
.

TomAFrench committed:
.

TomAFrench committed:
.

TomAFrench committed:
.

TomAFrench committed:
.

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix test case

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix noir build

guipublic committed:
remove sha256 blackbox tests

guipublic committed:
fix merge issue

guipublic committed:
fix unit test

guipublic committed:
update serialisation

guipublic committed:
fix warning

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
format fix

guipublic committed:
restore sha256 in stdlib

guipublic committed:
fix merge issue

guipublic committed:
fix merge issue

guipublic committed:
update yarn.lock

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

fcarreiro committed:
wip

fcarreiro committed:
wip

fcarreiro committed:
wip

guipublic committed:
remove sha256 avm tests

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
comment sha256compression in hashing.ts

fcarreiro committed:
fix hashing.ts and other cpp things

fcarreiro committed:
fix sizes in wire format

fcarreiro committed:
fix execution test

fcarreiro committed:
Merge remote-tracking branch 'origin/fc/avm-set-variants' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
update serialization test case

guipublic committed:
update yarn.lock

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
fix test case serialisation

TomAFrench committed:
chore: fix duplicated import

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

guipublic committed:
Merge branch 'master' into gd/noir_issue_4330

fcarreiro committed:
Merge remote-tracking branch 'origin/master' into gd/noir_issue_4330

fcarreiro committed:
fix sha indirects in avm cpp

fcarreiro commented:
Woohoo, all green! For our info

```
Token::total_supply: bytecode is 604 bytes
Token::constructor: bytecode is 7121 bytes
Token::mint_public: bytecode is 1903 bytes
Token::public_get_name: bytecode is 507 bytes
Token::is_minter: bytecode is 628 bytes
Token::_reduce_total_supply: bytecode is 891 bytes
Token::get_admin: bytecode is 507 bytes
Token::transfer_public: bytecode is 5304 bytes
Token::burn_public: bytecode is 5118 bytes
Token::shield: bytecode is 6030 bytes
Token::public_get_symbol: bytecode is 507 bytes
Token::set_minter: bytecode is 776 bytes
Token::assert_minter_and_mint: bytecode is 1236 bytes
Token::set_admin: bytecode is 653 bytes
Token::mint_private: bytecode is 2617 bytes
Token::_increase_public_balance: bytecode is 1087 bytes
Token::complete_refund: bytecode is 2330 bytes
Token::public_get_decimals: bytecode is 522 bytes
Token::balance_of_public: bytecode is 710 bytes
```

fcarreiro commented:
Posting all the others for the record. Also I just added the "bench-all" label so that it refreshes the benchmarks.

```
AuthWitTest::consume_public: bytecode is 510 bytes
AuthRegistry::is_reject_all: bytecode is 581 bytes
AuthRegistry::set_authorized: bytecode is 465 bytes
AuthRegistry::set_reject_all: bytecode is 364 bytes
AuthRegistry::consume: bytecode is 2637 bytes
AuthRegistry::_set_authorized: bytecode is 493 bytes
AuthRegistry::is_consumable: bytecode is 682 bytes
AvmInitializerTest::constructor: bytecode is 4578 bytes
AvmInitializerTest::read_storage_immutable: bytecode is 492 bytes
Auth::get_scheduled_authorized: bytecode is 595 bytes
Auth::get_authorized: bytecode is 697 bytes
Auth::set_authorized: bytecode is 4338 bytes
Auth::set_authorized_delay: bytecode is 4170 bytes
Auth::get_authorized_delay: bytecode is 952 bytes
Auth::constructor: bytecode is 4601 bytes
AvmTest::nested_static_call_to_set_storage: bytecode is 706 bytes
AvmTest::get_version: bytecode is 93 bytes
AvmTest::assert_timestamp: bytecode is 112 bytes
AvmTest::nullifier_collision: bytecode is 95 bytes
AvmTest::pedersen_hash: bytecode is 257 bytes
AvmTest::debug_logging: bytecode is 1492 bytes
AvmTest::set_opcode_u64: bytecode is 100 bytes
AvmTest::get_address: bytecode is 93 bytes
AvmTest::check_selector: bytecode is 108 bytes
AvmTest::assert_same: bytecode is 120 bytes
AvmTest::create_different_nullifier_in_nested_call: bytecode is 724 bytes
AvmTest::variable_base_msm: bytecode is 475 bytes
AvmTest::get_function_selector: bytecode is 93 bytes
AvmTest::emit_unencrypted_log: bytecode is 2636 bytes
AvmTest::get_sender: bytecode is 93 bytes
AvmTest::get_l2_gas_left: bytecode is 93 bytes
AvmTest::get_timestamp: bytecode is 93 bytes
AvmTest::create_same_nullifier_in_nested_call: bytecode is 713 bytes
AvmTest::bulk_testing: bytecode is 20896 bytes
AvmTest::keccak_hash: bytecode is 3721 bytes
AvmTest::assertion_failure: bytecode is 103 bytes
AvmTest::add_u128: bytecode is 208 bytes
AvmTest::sha256_hash: bytecode is 4062 bytes
AvmTest::get_fee_per_da_gas: bytecode is 93 bytes
AvmTest::set_opcode_really_big_field: bytecode is 124 bytes
AvmTest::to_radix_le: bytecode is 254 bytes
AvmTest::get_da_gas_left: bytecode is 93 bytes
AvmTest::get_block_number: bytecode is 93 bytes
AvmTest::return_oracle: bytecode is 347 bytes
AvmTest::test_get_contract_instance: bytecode is 903 bytes
AvmTest::note_hash_exists: bytecode is 124 bytes
AvmTest::test_get_contract_instance_raw: bytecode is 399 bytes
AvmTest::get_fee_per_l2_gas: bytecode is 93 bytes
AvmTest::read_storage_list: bytecode is 626 bytes
AvmTest::keccak_f1600: bytecode is 484 bytes
AvmTest::new_note_hash: bytecode is 89 bytes
AvmTest::l1_to_l2_msg_exists: bytecode is 124 bytes
AvmTest::add_storage_map: bytecode is 917 bytes
AvmTest::set_opcode_u32: bytecode is 96 bytes
AvmTest::set_read_storage_single: bytecode is 594 bytes
AvmTest::new_nullifier: bytecode is 89 bytes
AvmTest::set_storage_list: bytecode is 262 bytes
AvmTest::emit_nullifier_and_check: bytecode is 130 bytes
AvmTest::u128_from_integer_overflow: bytecode is 746 bytes
AvmTest::nullifier_exists: bytecode is 124 bytes
AvmTest::nested_call_to_add: bytecode is 993 bytes
AvmTest::pedersen_commit: bytecode is 383 bytes
AvmTest::assert_nullifier_exists: bytecode is 124 bytes
AvmTest::read_storage_single: bytecode is 445 bytes
AvmTest::set_storage_single: bytecode is 241 bytes
AvmTest::get_args_hash: bytecode is 2197 bytes
AvmTest::set_storage_map: bytecode is 447 bytes
AvmTest::modulo2: bytecode is 110 bytes
AvmTest::elliptic_curve_add_and_double: bytecode is 182 bytes
AvmTest::u128_addition_overflow: bytecode is 1376 bytes
AvmTest::add_args_return: bytecode is 105 bytes
AvmTest::get_storage_address: bytecode is 93 bytes
AvmTest::set_opcode_u8: bytecode is 92 bytes
AvmTest::assert_calldata_copy: bytecode is 465 bytes
AvmTest::pedersen_hash_with_index: bytecode is 257 bytes
AvmTest::get_transaction_fee: bytecode is 93 bytes
AvmTest::set_opcode_small_field: bytecode is 108 bytes
AvmTest::read_storage_map: bytecode is 566 bytes
AvmTest::get_chain_id: bytecode is 93 bytes
AvmTest::nested_static_call_to_add: bytecode is 993 bytes
AvmTest::poseidon2_hash: bytecode is 1515 bytes
AvmTest::nested_call_to_add_with_gas: bytecode is 996 bytes
AvmTest::set_opcode_big_field: bytecode is 124 bytes
AvmTest::send_l2_to_l1_msg: bytecode is 99 bytes
EasyPrivateVoting::add_to_tally_public: bytecode is 1110 bytes
EasyPrivateVoting::constructor: bytecode is 4891 bytes
EasyPrivateVoting::end_vote: bytecode is 652 bytes
Claim::constructor: bytecode is 5110 bytes
Delegator::public_delegate_set_value: bytecode is 524 bytes
ImportTest::pub_call_public_fn: bytecode is 706 bytes
PriceFeed::get_price: bytecode is 603 bytes
PriceFeed::set_price: bytecode is 452 bytes
FeeJuice::_increase_public_balance: bytecode is 1040 bytes
FeeJuice::balance_of_public: bytecode is 663 bytes
FeeJuice::check_balance: bytecode is 923 bytes
FeeJuice::set_portal: bytecode is 1042 bytes
FPC::pay_refund_with_shielded_rebate: bytecode is 1708 bytes
FPC::pay_refund: bytecode is 1708 bytes
FPC::constructor: bytecode is 4579 bytes
FPC::prepare_fee: bytecode is 1290 bytes
Router::_check_timestamp: bytecode is 822 bytes
Router::_check_block_number: bytecode is 805 bytes
Child::set_value_twice_with_nested_last: bytecode is 950 bytes
Child::pub_inc_value_internal: bytecode is 714 bytes
Child::set_value_twice_with_nested_first: bytecode is 950 bytes
Child::pub_inc_value: bytecode is 686 bytes
Child::pub_get_value: bytecode is 146 bytes
Child::set_value_with_two_nested_calls: bytecode is 776 bytes
Child::pub_set_value: bytecode is 337 bytes
DelegatedOn::public_set_value: bytecode is 251 bytes
PrivateFPC::constructor: bytecode is 4595 bytes
Parent::public_static_call: bytecode is 614 bytes
Parent::public_nested_static_call: bytecode is 4065 bytes
Parent::pub_entry_point_twice: bytecode is 745 bytes
Parent::pub_entry_point: bytecode is 457 bytes
Lending::get_position: bytecode is 4157 bytes
Lending::get_assets: bytecode is 871 bytes
Lending::deposit_public: bytecode is 1981 bytes
Lending::_borrow: bytecode is 11002 bytes
Lending::withdraw_public: bytecode is 1091 bytes
Lending::borrow_public: bytecode is 1091 bytes
Lending::repay_public: bytecode is 1810 bytes
Lending::update_accumulator: bytecode is 11588 bytes
Lending::_repay: bytecode is 6683 bytes
Lending::_deposit: bytecode is 1253 bytes
Lending::init: bytecode is 1525 bytes
Lending::_withdraw: bytecode is 10905 bytes
Lending::get_asset: bytecode is 833 bytes
Crowdfunding::_publish_donation_receipts: bytecode is 850 bytes
Crowdfunding::init: bytecode is 5653 bytes
Benchmarking::broadcast: bytecode is 627 bytes
Benchmarking::increment_balance: bytecode is 1405 bytes
StaticParent::public_call: bytecode is 457 bytes
StaticParent::public_nested_static_call: bytecode is 1322 bytes
StaticParent::public_get_value_from_child: bytecode is 822 bytes
StaticParent::public_static_call: bytecode is 614 bytes
AppSubscription::constructor: bytecode is 6703 bytes
Uniswap::_approve_bridge_and_exit_input_asset_to_L1: bytecode is 8912 bytes
Uniswap::constructor: bytecode is 4579 bytes
Uniswap::_assert_token_is_same: bytecode is 540 bytes
Uniswap::swap_public: bytecode is 16543 bytes
NFT::set_admin: bytecode is 653 bytes
NFT::is_minter: bytecode is 628 bytes
NFT::constructor: bytecode is 6578 bytes
NFT::mint: bytecode is 1478 bytes
NFT::_store_point_in_transient_storage: bytecode is 366 bytes
NFT::get_admin: bytecode is 507 bytes
NFT::_finish_transfer_to_public: bytecode is 427 bytes
NFT::transfer_in_public: bytecode is 4330 bytes
NFT::public_get_name: bytecode is 507 bytes
NFT::owner_of: bytecode is 947 bytes
NFT::public_get_symbol: bytecode is 507 bytes
NFT::set_minter: bytecode is 776 bytes
NFT::finalize_transfer_to_private: bytecode is 2210 bytes
TokenBridge::exit_to_l1_public: bytecode is 8766 bytes
TokenBridge::_call_mint_on_token: bytecode is 1290 bytes
TokenBridge::_assert_token_is_same: bytecode is 532 bytes
TokenBridge::claim_public: bytecode is 16138 bytes
TokenBridge::get_portal_address_public: bytecode is 550 bytes
TokenBridge::constructor: bytecode is 4733 bytes
TokenBridge::get_token: bytecode is 507 bytes
Spam::public_spam: bytecode is 1050 bytes
StaticChild::pub_set_value: bytecode is 337 bytes
StaticChild::pub_get_value: bytecode is 161 bytes
StaticChild::pub_illegal_inc_value: bytecode is 701 bytes
StaticChild::pub_inc_value: bytecode is 686 bytes
InclusionProofs::push_nullifier_public: bytecode is 136 bytes
InclusionProofs::test_nullifier_inclusion_from_public: bytecode is 166 bytes
InclusionProofs::constructor: bytecode is 4195 bytes
TestLog::emit_unencrypted_events: bytecode is 1343 bytes
DocsExample::spend_public_authwit: bytecode is 98 bytes
DocsExample::initialize_public_immutable: bytecode is 792 bytes
DocsExample::get_shared_immutable_constrained_public_indirect: bytecode is 511 bytes
DocsExample::update_leader: bytecode is 302 bytes
DocsExample::get_shared_immutable_constrained_public: bytecode is 516 bytes
DocsExample::initialize_shared_immutable: bytecode is 792 bytes
DocsExample::get_shared_immutable_constrained_public_multiple: bytecode is 721 bytes
Test::is_time_equal: bytecode is 122 bytes
Test::create_l2_to_l1_message_arbitrary_recipient_public: bytecode is 99 bytes
Test::emit_nullifier_public: bytecode is 89 bytes
Test::emit_unencrypted: bytecode is 1330 bytes
Test::dummy_public_call: bytecode is 105 bytes
Test::consume_mint_public_message: bytecode is 14845 bytes
Test::consume_message_from_arbitrary_sender_public: bytecode is 9116 bytes
Test::create_l2_to_l1_message_public: bytecode is 183 bytes
Test::assert_public_global_vars: bytecode is 252 bytes
StatefulTest::public_constructor: bytecode is 4649 bytes
StatefulTest::increment_public_value_no_init_check: bytecode is 696 bytes
StatefulTest::increment_public_value: bytecode is 743 bytes
StatefulTest::get_public_value: bytecode is 566 bytes
TokenBlacklist::transfer_public: bytecode is 6718 bytes
TokenBlacklist::update_roles: bytecode is 5455 bytes
TokenBlacklist::constructor: bytecode is 8346 bytes
TokenBlacklist::_reduce_total_supply: bytecode is 891 bytes
TokenBlacklist::balance_of_public: bytecode is 710 bytes
TokenBlacklist::mint_private: bytecode is 2928 bytes
TokenBlacklist::total_supply: bytecode is 604 bytes
TokenBlacklist::mint_public: bytecode is 2845 bytes
TokenBlacklist::get_roles: bytecode is 960 bytes
TokenBlacklist::burn_public: bytecode is 5906 bytes
TokenBlacklist::_increase_public_balance: bytecode is 1087 bytes
TokenBlacklist::shield: bytecode is 6818 bytes
CardGame::on_cards_claimed: bytecode is 3814 bytes
CardGame::on_card_played: bytecode is 4675 bytes
CardGame::start_game: bytecode is 6223 bytes
CardGame::on_game_joined: bytecode is 3550 bytes
Token::burn_public: bytecode is 5118 bytes
Token::mint_public: bytecode is 1903 bytes
Token::assert_minter_and_mint: bytecode is 1236 bytes
Token::shield: bytecode is 6030 bytes
Token::mint_private: bytecode is 2617 bytes
Token::get_admin: bytecode is 507 bytes
Token::complete_refund: bytecode is 2330 bytes
Token::public_get_decimals: bytecode is 522 bytes
Token::constructor: bytecode is 7121 bytes
Token::public_get_name: bytecode is 507 bytes
Token::_reduce_total_supply: bytecode is 891 bytes
Token::public_get_symbol: bytecode is 507 bytes
Token::total_supply: bytecode is 604 bytes
Token::set_admin: bytecode is 653 bytes
Token::set_minter: bytecode is 776 bytes
Token::transfer_public: bytecode is 5304 bytes
Token::balance_of_public: bytecode is 710 bytes
Token::_increase_public_balance: bytecode is 1087 bytes
Token::is_minter: bytecode is 628 bytes
```
